docs/rfq-and-price-oracle.md

```diff
@@ -0,0 +1,233 @@
+# Asset Decimal Display
+
+Within the Taproot Assets Protocol, an asset's unit is an integer (`uint64`).
+That means, the protocol cannot represent fractions of an asset.
+Therefore, any asset that represents a fiat currency would need to issue assets
+equivalent to at least the smallest unit in use.
+For example, an asset that represents the US-Dollar would need to be minted in a
+way that one asset unit would represent one USD **cent**.
+Or in other words, 100 units of such an asset would represent 1 US-Dollar.
+
+Because wallet software user interfaces aren't expected to know what
+"resolution" or precision any asset in the wild represents, a new field called
+`decimal_display` was added to the JSON metadata of new assets minted with
+`tapd v0.4.0-alpha` and later (this field is encoded in the metadata field as a
+JSON field, therefore it is only compatible with assets that have a JSON
+metadata field).
+
+An issuer can specify `tapcli assets mint --decimal_display X` to specify the
+number of decimal places the comma should be shifted to the left when displaying
+a sum of asset units.
+
+For the example above, a USD asset would choose `--decimal_display 2` to
+indicate that 100 units ($10^2$) should be displayed as `$1.00` in the wallet
+UI. Or another example: 1234 USD cent asset units would be displayed as `$12.34`
+in the UI.
+
+## Precision requirement for assets in the Lightning Network
+
+Due to the non-divisible (integer) nature of Taproot Asset units, the smallest
+asset amount that can be transferred with a Lightning Network HTLC is one asset
+unit.
+
+If one such asset unit represents significantly more than one satoshi, then in
+some cases, due to integer division and rounding up, the user might end up
+spending more assets than necessary when paying an invoice.
+
+**Example 1: Paying a 1 satoshi invoice**:
+
+While writing this article, one USD cent is roughly equivalent to 19 satoshi.
+
+So if a user with USD cent assets in their wallet attempted to pay an invoice
+denominated over 1 satoshi, they would need to send a full USD cent to satisfy
+the invoice (again, only full asset units can be transported over an HTLC).
+
+Even if one cent isn't much, the overpayment would still be roughly `19x`.
+
+**Example 2: Paying an invoice with MPP**:
+
+Multi-Part Payments (MPP) allow a single payment to be split up into multiple
+parts/paths, resulting in multiple HTLCs per payment.
+
+If a user wants to pay an invoice over `1,000,000` satoshi, that would be
+equivalent to `52,631.5789` USD cents.
```

Comment:
I didn't want to go into too much details re decimal display here. The goal was to show how using just a representation of cents would lead to quite a bit of overpayment in MPP. The 4 decimal places in the values is just so the example itself wouldn't suffer from loss of precision.